### PR TITLE
xds/resolver: Add missing double quote in json SC

### PR DESCRIPTION
**Pull request by easwars**

### CODE REVIEW

**Comment by easwars** [comment]
I have made changes to the test as well, but it includes some refactoring/cleanup etc. So, I want to do that as a separate PR.

File: xds/internal/resolver/xds_resolver.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ const jsonFormatSC = `{
     "loadBalancingConfig":[
       {
         "experimental_cds":{
-          "Cluster": %s
+          "Cluster": "%s"
```
Comment (menghanl) [inline xds/internal/resolver/xds_resolver.go]: Does `json.Unmarshal` fail without the quotes? Let's add some sanity check.

If you are adding those in the following PR, it's OK to submit this as is.

Comment (easwars) [reply xds/internal/resolver/xds_resolver.go]: Yes `json.Unmarshal` fails without the quotes. In the other test PR, I verify that the service config is parsed without an error by the cdsBuilder. It is an indirect test though. There doesn't seem to be an easy way to directly test if `jsonFormatSC` contains a valid JSON.

Specifically, handling this TODO was how I uncovered this issue:
https://github.com/grpc/grpc-go/blob/master/xds/internal/resolver/xds_resolver_test.go#L74

Comment (easwars) [reply xds/internal/resolver/xds_resolver.go]: The test cleanup PR is here:
https://github.com/grpc/grpc-go/pull/3233

And it fails as follows without the fix in here::

    easwars@golang:~/src/grpc-go$ go test -count=1 -race -cpu 4 -timeout=50s 
    google.golang.org/grpc/xds/internal/resolver/...
    --- FAIL: TestXDSResolverGoodServiceUpdate (0.00s)
        xds_resolver_test.go:317: ClientConn.UpdateState received error in service config: invalid character 'c' looking for beginning of value
    FAIL
    FAIL    google.golang.org/grpc/xds/internal/resolver    1.203s

**Pull request merged**